packages/nodes-base/nodes/Google/MyBusiness/PostDescription.ts

```diff
@@ -0,0 +1,1083 @@
+import {
+	NodeApiError,
+	NodeOperationError,
+	type IDataObject,
+	type IExecuteSingleFunctions,
+	type IN8nHttpFullResponse,
+	type INodeExecutionData,
+	type INodeProperties,
+	type JsonObject,
+} from 'n8n-workflow';
+
+import { addUpdateMaskPresend, handleDatesPresend, handlePagination } from './GenericFunctions';
+
+export const postOperations: INodeProperties[] = [
+	{
+		displayName: 'Operation',
+		name: 'operation',
+		type: 'options',
+		default: 'create',
+		noDataExpression: true,
+		displayOptions: { show: { resource: ['post'] } },
+		options: [
+			{
+				name: 'Create',
+				value: 'create',
+				action: 'Create post',
+				description: 'Create a new post on Google My Business',
+				routing: {
+					send: { preSend: [handleDatesPresend] },
+					request: {
+						method: 'POST',
+						url: '=/{{$parameter["account"]}}/{{$parameter["location"]}}/localPosts',
+					},
+				},
+			},
+			{
+				name: 'Delete',
+				value: 'delete',
+				action: 'Delete post',
+				description: 'Delete an existing post',
+				routing: {
+					request: {
+						method: 'DELETE',
+						url: '=/{{$parameter["post"]}}',
+						ignoreHttpStatusErrors: true,
+					},
+					output: {
+						postReceive: [
+							async function (
+								this: IExecuteSingleFunctions,
+								data: INodeExecutionData[],
+								response: IN8nHttpFullResponse,
+							): Promise<INodeExecutionData[]> {
+								if (response.statusCode < 200 || response.statusCode >= 300) {
+									const post = this.getNodeParameter('post', undefined) as IDataObject;
+									if (post && response.statusCode === 404) {
+										// Don't return a 404 error if the post does not exist
+										throw new NodeOperationError(
+											this.getNode(),
+											'The post you are deleting could not be found. Adjust the "post" parameter setting to delete the post correctly',
+										);
+									}
+
+									throw new NodeApiError(this.getNode(), response.body as JsonObject, {
+										message: response.statusMessage,
+										httpCode: response.statusCode.toString(),
+									});
+								}
+								return data;
+							},
+						],
+					},
+				},
+			},
+			{
+				name: 'Get',
+				value: 'get',
+				action: 'Get post',
+				description: 'Retrieve details of a specific post',
+				routing: {
+					request: {
+						method: 'GET',
+						url: '=/{{$parameter["post"]}}',
+						ignoreHttpStatusErrors: true,
+					},
+					output: {
+						postReceive: [
+							async function (
```

Comment:
Can you extract this to GenericFunctions?

Reply:
I have extracted it.